### PR TITLE
[Shaan] [Iot Device Catalogue] | Feature

### PR DESCRIPTION
**Pull request by ShaanCoding**
## Summary

This PR is the initial chunk of code regarding the IoT Device Catalogue Feature, this feature contains all CRUD operations related to the creation of devices

### CODE REVIEW

File: packages/backend/prisma/schema.prisma [hunk 1]
```diff
@@ -67,6 +67,7 @@ model Product {
   OrderLineItem OrderLineItem[]
   Category      ProductCategory? @relation(fields: [categoryId], references: [categoryId])
   categoryId    String?
+  lastUpdated   DateTime         @default(now())
```
Comment (sebasptsch) [inline packages/backend/prisma/schema.prisma:70]: Use prisma `@updatedAt`

File: packages/backend/src/controllers/categories.controller.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+  if (!deletedCategory) {
+    return reply.internalServerError("Failed to delete category");
+  }
+
+  return reply.status(200).send(deletedCategory);
+};
+
+// deleteCategories (DELETE) /
+export const deleteCategories = async (
```
Comment (sebasptsch) [inline packages/backend/src/controllers/categories.controller.ts:138]: Is there no schema for this request?

File: packages/backend/src/controllers/products.controller.ts [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+      category: true,
+      categoryId: true,
+      lastUpdated: true,
+    },
+  });
+
+  return reply.status(200).send(products);
+};
+
+interface ICreateProduct {
```
Comment (sebasptsch) [inline packages/backend/src/controllers/products.controller.ts:68]: this should be a schema

File: packages/backend/src/controllers/products.controller.ts [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+      image: true,
+      category: true,
+      lastUpdated: true,
+    },
+  });
+
+  return reply.status(200).send(product);
+};
+
+interface IDeleteProducts {
```
Comment (sebasptsch) [inline packages/backend/src/controllers/products.controller.ts:173]: Should be a schema

File: packages/backend/src/controllers/products.controller.ts [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+      productId,
+    },
+    data: {
+      name,
+      price,
+      stock,
+      description,
+      image,
+      category,
+      lastUpdated: new Date(),
```
Comment (sebasptsch) [inline packages/backend/src/controllers/products.controller.ts:268]: Prisma has inbuilt updatedAt

File: packages/frontend/src/App.tsx [hunk 6]
```diff
@@ -30,6 +37,21 @@ const router = createBrowserRouter([
         index: true,
         element: <Landing />,
       },
+      {
+        path: "/staff/inventory/manage",
+        loader: profileLoader(queryClient),
```
Comment (sebasptsch) [inline packages/frontend/src/App.tsx:42]: you can create a parent route that just uses the loader once. e.g. 
```tsx
{
  path: "/staff"
  loader: profileLoader(queryClient),
}
```
Child routes will get data from their nearest parent loader.

File: packages/frontend/src/features/IoTDeviceCatalogue/ProductTable.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+interface IProductTableProps {
+  isLightMode: boolean;
+  getProducts: {
+    data: Array<ProductsSchema>;
+  };
+  selectedItems: string[];
+  setSelectedItems: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+const ProductTable: React.FC<IProductTableProps> = ({
```
Comment (sebasptsch) [inline packages/frontend/src/features/IoTDeviceCatalogue/ProductTable.tsx:23]: have a look at `@tanstack/table`. We can make a generic datatable to use across all components:
See example [here](https://github.com/Team3132/AttendanceSystem/blob/master/packages/frontend/src/components/DataTable.tsx)

**Review by sebasptsch — CHANGES_REQUESTED** [review]
There's a few things to address.

File: packages/backend/prisma/schema.prisma (hunk 1 above)
Comment (ShaanCoding) [reply packages/backend/prisma/schema.prisma:70]: Done

File: packages/backend/src/controllers/products.controller.ts (hunk 5 above)
Comment (ShaanCoding) [reply packages/backend/src/controllers/products.controller.ts:268]: Done

File: packages/frontend/src/features/IoTDeviceCatalogue/ProductTable.tsx (hunk 7 above)
Comment (sebasptsch) [reply packages/frontend/src/features/IoTDeviceCatalogue/ProductTable.tsx:23]: defer

**Review by sebasptsch — APPROVED** [review]

**Comment by ShaanCoding** [comment]
image

**Pull request merged**